src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,277 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Scanner;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+
+import duke.storage.Storage;
+
+/**
+ * Is the main class of the program.
+ */
+public class Duke extends Application {
+
+    final TaskList taskList;
+    final Storage storage;
+    final Parser parser;
+    final Ui ui;
+
+    /**
+     * Constructs an instance of Duke.
+     */
+    public Duke() {
+        taskList = new TaskList();
+        storage = new Storage(taskList);
+        parser = new Parser();
+        ui = new Ui();
+    }
+
+    private static void assertThis(boolean expectsTrue, String failureMessage) throws DukeException {
+        if (!expectsTrue) {
+            throw new DukeException(failureMessage);
+        }
+    }
+
+    private void displayTaskCount() {
+        if (taskList.isEmpty()) {
+            ui.println("You do not have any task!");
+        } else {
+            ui.println("Now you have " + taskList.size() + " task(s) in the list.");
+        }
+    }
+
+    private void displayTasks() {
+        if (taskList.size() == 0) {
+            ui.println("Your list is empty.");
+        } else {
+            ui.println("You have the following task(s):");
+            for (int i = 0; i < taskList.size(); i++) {
+                ui.println("\t" + (i + 1) + ". " + taskList.get(i));
+            }
+        }
+    }
+
+    private void addNewTask(Task task) {
+        taskList.add(task);
+        ui.println("Got it. I've added this task:");
+        ui.println("\t" + task);
+
+        storage.saveDataToFile();
+        displayTaskCount();
+    }
+
+    void cli() {
```

Comment:
I feel like this method is too long and has a lot of nesting, perhaps the different commands could be extracted out into separate methods.

Reply:
Thank you for your feedback! I will make changes accordingly.